pkg/kubectl/cmd/delete.go

```diff
@@ -91,13 +92,22 @@ var (
 type DeleteOptions struct {
 	resource.FilenameOptions
 
-	Selector        string
-	DeleteAll       bool
-	IgnoreNotFound  bool
-	Cascade         bool
-	DeleteNow       bool
-	ForceDeletion   bool
+	Selector       string
+	DeleteAll      bool
+	IgnoreNotFound bool
+	Cascade        bool
+	DeleteNow      bool
+	ForceDeletion  bool
+	// Whether we should wait for resource deletion.
+	// Tied to `--wait` flag.
 	WaitForDeletion bool
```

Comment:
We should only have one field in the options… handle the intent-checking outside, but only make the delete operation need to know whether it should wait or not